src/contracts/IBridge.ts

```diff
@@ -1,13 +1,14 @@
-import { Contract, EventData } from 'web3-eth-contract';
+import { Contract, EventLog } from 'web3-eth-contract';
+import {ContractAbi} from "web3";
 
 export interface IBridge {
-  bridgeContract: Contract;
+  bridgeContract: Contract<ContractAbi>;
 
   getFederation();
 
   getAllowedTokens();
 
-  getPastEvents(eventName: string, destinationChainId: number, options: any): Promise<EventData[]>;
+  getPastEvents(eventName: string, destinationChainId: number, options: any): Promise<(string | EventLog)[]>;
```

Comment:
This interface is way different than string, are you sure that the implementation can return both? In which scenario it returns each type? Perhaps is worth to test and review that to keep either string or EventLog only

Reply:
(The same comment in general for other parts of the PR that have this union)